### ISSUE TITLE
[RFC]: Remove `Layout` and `ItemLayout` components and references

### ISSUE THREAD

**Issue by bmdalex** [microsoft/fluent-ui-react#525]
### Problem description
We have quite a few issues with the `Layout` and `ItemLayout` components, so I'm opening the following question:

#### Do we still need `Layout` and `ItemLayout`?

The cons seem to trump the pros;

#### Pros:
- out of the box solution for styling parts of components when it comes to positioning items using flex-box

#### Cons:
- renders big amount of DOM nodes: #241 (DOM tree reduced by 50% for `Chat.Message` in #518 )
- the components do not adhere to Stardust principles:
  - slots are not shorthands;
  - styling is done incorrectly -> user cannot target component slots for styling: #403
- they need a big refactoring (started in #236)
- they are not used that much internally and probably externally as well (need to confirm that)
- possible better solution is the `template` prop approach described in #401 
- buggy: see #403, #477

Therefore, we're removing their usage from components such as:

- [x] `Chat.Message` - #518 
- [ ] `Label`
- [ ] `ListItem` - fixes #403 

#### This fixes #241, fixes #403, fixes #477, fixes #467

**Comment by levithomason** [comment]
@Bugaa92 I don't think you can close an issue from the description of another issue.  The "fixes" statement at the bottom here likely needs to be moved to a PR.
